src/main/java/com/atlassian/db/replica/spi/circuitbreaker/CircuitBreaker.java

```diff
@@ -1,9 +1,18 @@
 package com.atlassian.db.replica.spi.circuitbreaker;
 
-import com.atlassian.db.replica.api.circuitbreaker.BreakerState;
+import com.atlassian.db.replica.api.SqlCall;
+
+import java.sql.SQLException;
 
 public interface CircuitBreaker {
-    BreakerState getState();
 
-    void handle(Throwable throwable);
+    boolean canCall();
```

Comment:
It may be misleading. We don't call this method when we decide if we want to call DB or not. It's rather `canEstablishConnectionToReplica`. 🤔

Reply:
Actually, it's `canUseDualConnection`

Reply:
It will still use DualConnection's that are in the fly. It will not create new ones. `canCreateDualConnection`?